src/main/java/frc/robot/subsystems/intake/IntakeSubsystem.java

```diff
@@ -0,0 +1,533 @@
+        },
+        () -> {});
+  }
+
+  public Command algaeIntake() {
+    return startEnd(
+        () -> {
+          goToPivotPosition(kalgae);
+          intakeMotor.set(-kIntakeSpeed);
+        },
+        () -> {});
+  }
+
+  public Command algaeHold() {
+    return startEnd(
+        () -> {
+          goToPivotPosition(kalgae);
+          intakeMotor.set(-kIntakeSpeed);
+        },
+        () -> {});
+  }
+
+  public Command algaeOuttake() {
+    return startEnd(
+        () -> {
+          goToPivotPosition(kalgae);
+          intakeMotor.set(kIntakeSpeed);
+        },
+        () -> {});
+  }
+
+  public Command l1ScoreModeA() {
+    return startEnd(
+        () -> {
+          goToPivotPosition(kl1);
+          intakeMotor.set(kIntakeSpeed);
+        },
+        () -> {});
+  }
+
+  public Command l1ScoreModeB() {
+    return startEnd(
+        () -> {
+          goToPivotPosition(kPivotRetractAngle);
+          intakeMotor.set(kIntakeSpeed / 5);
+          conveyorMotor.set(kConveyorSpeed);
+        },
+        () -> {});
+  }
+
+  public Command Idle() {
+    return startEnd(
+        () -> {
+          goToPivotPosition(kPivotRetractAngle);
+          setIntakeMotor(0);
+          setConveyerMotor(0);
+        },
+        () -> {});
+  }
+
+  @Override
+  public void periodic() {
+    // This method will be called once per scheduler run
+    SmartDashboard.putNumber("Intake/Intake Motor Output", intakeMotor.get());
+    SmartDashboard.putNumber("Intake/Pivot Motor Output", pivotMotor.get());
+    SmartDashboard.putNumber("Intake/Conveyor Motor Output", conveyorMotor.get());
+    SmartDashboard.putNumber("Intake/Pivot Setpoint", pivotSetpoint.in(Degrees));
+    SmartDashboard.putNumber(
+        "Intake/Pivot Position (Degrees)", pivotMotor.getPosition().getValue().in(Degrees));
+    SmartDashboard.putNumber(
+        "Intake/Absolute Encoder Position (Rotation)",
+        Rotations.of(throughBoreEncoder.get()).in(Degrees));
+    SmartDashboard.putString("Intake/Intake State", intakeState.toString());
+    pivotOutput.log(pivotMotor.get());
+    if (this.getCurrentCommand() != null) {
+      currentCommand.log(this.getCurrentCommand().getName());
+    } else {
+      currentCommand.log("none");
+    }
+  }
+
+  public void simulationPeriodic() {
+    pivotSim.setInputVoltage(pivotMotor.getMotorVoltage().getValue().in(Volts));
+    pivotSim.update(Robot.defaultPeriodSecs);
+    simPivotMotor.setRawRotorPosition(Radians.of(pivotSim.getAngleRads()).times(kGearing));
+    simPivotMotor.setRotorVelocity(
+        RadiansPerSecond.of(pivotSim.getVelocityRadPerSec()).times(kGearing));
+    simPivotMotor.setSupplyVoltage(RobotController.getBatteryVoltage());
+    pivotArmMech.setAngle(Radians.of(pivotSim.getAngleRads()).in(Degrees));
+
+    switch (intakeState) {
+      case IDLE:
+        RobotContainer.sensors.setSimState(CoralEnum.NO_CORAL);
+
+        if (atSetpoint(kPivotExtendAngle) && intakeMotor.get() == kIntakeSpeed) {
```

Comment:
Turns out this doesn't work. For some reason get() doesn't return the same value set using set() in simulation. 
We'll have to rethink this.